Resolve bug with torch.multiprocessing.spawn when it hangs during huge exception stacktrace

aivanou opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#46754 Resolve bug with torch.multiprocessing.spawn when it hangs during huge exception stacktrace**

Differential Revision: [D24568103](https://our.internmc.facebook.com/intern/diff/D24568103)

aivanou committed:
The diff resolves bug with torch.mulpiprocessing.spawn where the worker process has huge exception stacktrace. The module uses multiprocessing.queue to transfer stacktrace between child and parent processes. The mp.queue module uses pipes underneath. If the writer fills the pipe and the reader process will not start consuming it, the writer process will hang forever. This diff changes the way the torch.multiprocessing.spawn waits for processes to finish.

[ghstack-poisoned]

tierex commented:
The diff resolves bug with torch.mulpiprocessing.spawn where the worker process has huge exception stacktrace. The module uses multiprocessing.queue to transfer stacktrace between child and parent processes. The mp.queue module uses pipes underneath. If the writer fills the pipe and the reader process will not start consuming it, the writer process will hang forever. This diff changes the way the torch.multiprocessing.spawn waits for processes to finish.

aivanou committed:
Update on "Resolve bug with torch.multiprocessing.spawn when it hangs during huge exception stacktrace"




[ghstack-poisoned]

aivanou committed:
Update on "Resolve bug with torch.multiprocessing.spawn when it hangs during huge exception stacktrace"




[ghstack-poisoned]

aivanou committed:
Update on "Resolve bug with torch.multiprocessing.spawn when it hangs during huge exception stacktrace"




[ghstack-poisoned]

aivanou committed:
Update on "Resolve bug with torch.multiprocessing.spawn when it hangs during huge exception stacktrace"


Differential Revision: [D24568103](https://our.internmc.facebook.com/intern/diff/D24568103)

[ghstack-poisoned]

aivanou committed:
Update on "Resolve bug with torch.multiprocessing.spawn when it hangs during huge exception stacktrace"


Differential Revision: [D24568103](https://our.internmc.facebook.com/intern/diff/D24568103)

[ghstack-poisoned]

ezyang commented:
Assigned for review based on sentinel, but smh I don't feel like the sentinel guess was very good. Seems like @taohe is probably the correct reviewer.

ailzhang commented:
Yea I'm probably not the best one to review this change, cc @VitalyFedyunin here :D .

aivanou committed:
Update on "Resolve bug with torch.multiprocessing.spawn when it hangs during huge exception stacktrace"


Differential Revision: [D24568103](https://our.internmc.facebook.com/intern/diff/D24568103)

[ghstack-poisoned]
